compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/lower_to_aie.mlir

```diff
@@ -665,7 +668,7 @@ module attributes {hal.executable.target = #executable_target_amdaie_xclbin_fb}
 // CHECK-SAME:          metadata = @[[OBJ2]]
 // CHECK-SAME:          memref<32x32xf32>
 // CHECK:           aiex.npu.dma_memcpy_nd
-// CHECK-SAME:          %[[RHS]][0, 0, 0, 0][1, 2, 32, 8][0, 8, 16, 1]
+// CHECK-SAME:          %[[RHS]][0, 0, 0, 34][1, 2, 32, 8][0, 8, 16, 1]
```

Comment:
can you explain how you are reaching 34?, in the base example, you have 
` [%c0, %c1, %c2] [%c2, %c32, %c16] [%c16, %c32, %c1] `
this to me means linear offset is, 1*32+2*16 = 48, so now since you are stepping 2 elements, you get 24?

Reply:
So, I tried the offset suggestion (as per https://github.com/nod-ai/iree-amd-aie/pull/573#discussion_r1684537348) and it worked.

The point you made for the linearized offset to be 17 makes sense to me. Good catch! :)
I've therefore added => `original linearized offset / upcastDivisor` as the linearized offset modification and I've added that as part of the current revision's push. It worked too.

But both the above point "worked" because we are dealing with a zero offset case. We cannot verify the same for a non-zero case just yet because of https://github.com/nod-ai/iree-amd-aie/pull/573#discussion_r1684159453.

I believe we should revisit/modify the logic for metadata later when we're able to successfully compile a non-zero offset case for bf16 as it'd help us in verifying the implementation (CC: @yzhang93 ).

And currently since we have our first bf16 Matmul case working (I again tried locally with the set of changes that went in recently and affected the input IR), it'd be nice to unblock getting this in, in my opinion.

But if we have a metadata algorithm that we know will work irrespective and there's some other way that we can verify  the same - I'd be happy to know and include the same in this PR.